packages/circuit-ui/components/NotificationInline/NotificationInline.tsx

```diff
@@ -0,0 +1,267 @@
+
+const TRANSITION_DURATION = 200;
+const DEFAULT_HEIGHT = 'auto';
+
+type Variant = 'info' | 'confirm' | 'notify' | 'alert';
+
+type Action = ButtonProps;
+
+type CloseProps =
+  | {
+      /**
+       * Renders a close button in the top right corner and calls the provided function
+       * when the button is clicked.
+       */
+      onClose: (event: ClickEvent) => void;
+      /**
+       * Text label for the close button for screen readers.
+       * Important for accessibility.
+       */
+      closeButtonLabel: string;
+    }
+  | { onClose?: never; closeButtonLabel?: never };
+
+export type BaseProps = HTMLAttributes<HTMLDivElement> & {
+  /**
+   * Notification inline variants. Defaults to `info`.
+   */
+  variant?: Variant;
+  /**
+   * Notification inline headline to provide information (optional)
+   */
+  headline?: string;
+  /**
+   * A body copy to provide information
+   */
+  body: string;
+  /**
+   * An optional call-to-action button.
+   */
+  action?: Action;
+  /**
+   * Whether the notification inline is visible.
+   */
+  isVisible?: boolean;
+  /**
+   * Additional data that is dispatched with the tracking event.
+   */
+  tracking?: TrackingProps;
+  /**
+   * A clear and concise description of the icon and the Toast's purpose. If the toast body is self-explanatory pass an empty string.
+   */
+  iconLabel?: string;
+};
+
+export type NotificationInlineProps = BaseProps & CloseProps;
+
+// TODO: update the design token colors to be info/confirm/alert/notify, then remove this mapping
+const colorMap = {
+  info: 'p500',
+  confirm: 'success',
+  alert: 'danger',
+  notify: 'warning',
+} as const;
+
+const iconMap = {
+  info: Info,
+  confirm: Confirm,
+  alert: Alert,
+  notify: Notify,
+};
+
+const inlineWrapperStyles = () => css`
+  display: inline-flex;
+  overflow: hidden;
+  will-change: height;
+  transition: opacity ${TRANSITION_DURATION}ms ease-in-out,
+    height ${TRANSITION_DURATION}ms ease-in-out,
+    visibility ${TRANSITION_DURATION}ms ease-in-out;
+`;
+
+const NotificationInlineWrapper = styled('div')(inlineWrapperStyles);
+
+type ContentWrapperProps = {
+  variant: Variant;
+};
+
+const contentWrapperStyles = ({
+  theme,
+  variant,
+}: ContentWrapperProps & StyleProps) => css`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  background-color: ${theme.colors.bodyBg};
+  padding: ${theme.spacings.kilo} ${theme.spacings.mega};
+  border-radius: ${theme.borderRadius.byte};
+  border: ${theme.borderWidth.mega} solid ${theme.colors[colorMap[variant]]};
+`;
+
+const ContentWrapper = styled('div')<ContentWrapperProps>(contentWrapperStyles);
+
+const contentStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding-right: ${theme.spacings.peta};
+  padding-left: ${theme.spacings.mega};
+`;
+
+const Content = styled('div')(contentStyles);
+
+const actionButtonStyles = ({ theme }: StyleProps & ButtonProps) =>
```

Comment:
Could we use a secondary button and remove its border instead of the custom button here? Then we wouldn't need to rewrite all of the state-specific styles

Reply:
Hmm, in either case I need to add the changes I have now, cause they are all the ones we need for the cta button, changing to bold, changing color and adding underline.

Reply:
Yes, but we wouldn't have to handle the `hover`, `active` and `focus` states. But I'm fine with keeping it as-is as well, doesn't make a huge difference. If we reuse the cta button elsewhere though I'd extract it into a component (an internal one)